Soft transitions

hpmax opened the pull request:
I thought I had already committed part of this, that included a number of efficiency improvements, but apparently, I either didn't or no one accepted the changes.

The main thing in this new release is soft volume transitions which should remove clicks and pops.

Things I'd still like to consider doing but haven't done are modified waveforms and and fixed point.  Also as previously pointed out, the pulseaudio system is screwed up as run from services.

hpmax committed:
Soft transitions

kedder commented on audiovario.c:
```diff
@@ -39,105 +39,141 @@ void vario_unmute(){
-	vario_config[vario].pulse_duty = PULSE_DUTY;
+	// vario_config[vario].pulse_length = PULSE_LENGTH; // don't need this any more (see loval)
+	// vario_config[vario].pulse_length_gain = PULSE_LENGTH_GAIN; // don't need this any more (see hival)
+	// vario_config[vario].pulse_duty = PULSE_DUTY; // don't need this any more (see riseduty + risedutyfall)
```
Please remove these lines if they are not needed

kedder reviewed (approved):
Looks good otherwise

hpmax committed:
Cleanup comments

hpmax commented:
Okay, cleaned up the comments a bit.

Many of the parameters that we input to control the behavior are no longer actually used by the program, instead I'm now using them to pre-compute derivatives of the information to speed up processing in the audio generation loop.  So pulse_length, and pulse_length_gain are not directly used, but they are displayed upon initial run.  With those lines commented out, IF the values aren't specified in the .conf file, they would be displayed incorrectly, even though it wouldn't impact the actual program behavior.  So I put them back in.  Pulse_length_duty on the other hand is never used except to calculate other values so I deleted that line.

kedder reviewed (approved):
LGTM
